lib/Backend/AsmJsJITInfo.cpp

```diff
@@ -159,6 +159,11 @@ AsmJsJITInfo::UsesHeapBuffer() const
 bool
 AsmJsJITInfo::AccessNeedsBoundCheck(uint offset) const
 {
+#ifdef _WIN32
     // Normally, heap has min size of 0x10000, but if you use ChangeHeap, min heap size is increased to 0x1000000
     return offset >= 0x1000000 || (IsHeapBufferConst() && offset >= 0x10000);
```

Comment:
Will revert this. (Looks I originally added this when my problem was the other piece not emitting bound check.)